pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -120,26 +121,93 @@ func (bw *quotaRestrictedBlobWriter) Commit(ctx context.Context, provisional dis
 	return can, err
 }
 
-// admitBlobWrite checks whether the blob does not exceed image quota, if set. Returns
+// admitBlobWrite checks whether the blob does not exceed image quota or limit ranges, if set. Returns
 // ErrAccessDenied error if the quota is exceeded.
-func admitBlobWrite(ctx context.Context, repo *repository) error {
-	rqs, err := repo.quotaClient.ResourceQuotas(repo.namespace).List(kapi.ListOptions{})
-	if err != nil {
-		if kerrors.IsForbidden(err) {
-			context.GetLogger(ctx).Warnf("Cannot list resourcequotas because of outdated cluster roles: %v", err)
-			return nil
-		}
-		context.GetLogger(ctx).Errorf("Failed to list resourcequotas: %v", err)
+func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
+	if err := admitLimitRanges(ctx, repo, size); err != nil {
 		return err
 	}
 
+	if err := admitQuotas(ctx, repo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// admitLimitRanges checks the blob against any established limit ranges.
+func admitLimitRanges(ctx context.Context, repo *repository, size int64) error {
+	if size < 1 {
+		return nil
+	}
+
+	var lrs *kapi.LimitRangeList
+	obj, exists, err := projectLimitRanges.Get(repo.namespace)
+	if !exists {
+		context.GetLogger(ctx).Debugf("Listing limit ranges in namespace %s", repo.namespace)
+		lrs, err = repo.limitClient.LimitRanges(repo.namespace).List(kapi.ListOptions{})
+		if err != nil {
+			if !repo.enforceQuota && kerrors.IsForbidden(err) {
+				context.GetLogger(ctx).Warnf("Cannot list limitranges because of outdated cluster roles: %v", err)
+				return nil
```

Comment:
Maybe I misunderstood. What `--enforce-quota=true` does is that the limits are checked and writes will be denied if exceeded or quotas cannot be obtained. `--enforce-quota=false` doesn't turn off quota checks. Writes will be still denied if an abuse is detected, however a failure to obtain a quota or limit object from etcd will be silently ignored and the write will be allowed.